### PR TITLE
halc: Create new package.

### PR DESCRIPTION
**Pull request by lingnanyuan**

### CODE REVIEW

File: var/spack/repos/builtin/packages/halc/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    """HALC is software that makes error correction for long reads with
+     high throughput."""
+
+    homepage = "https://github.com/lanl001/halc"
+    url      = "https://github.com/lanl001/halc/archive/v1.1.tar.gz"
+
+    version('1.1', '4b289b366f6a5400ca481993aa68dd9c')
+
+    depends_on('blasr')
+    depends_on('lordec')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/halc/package.py]: If you make blasr and lordec `run` dependencies, Spack should take care of updating `PATH`.

File: var/spack/repos/builtin/packages/halc/package.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    depends_on('dos2unix', type='build')
+    depends_on('python', type='run')
+
+    def build(self, spec, prefix):
+        make('all', parallel=False)
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/halc/package.py]: You can simply set `parallel = False` within the package and remove the build function.

File: var/spack/repos/builtin/packages/halc/package.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    def install(self, spec, prefix):
+        install_tree('.', prefix.bin)
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/halc/package.py]: This seems to be the cause of `prefix.bin.bin` below. Something like
```python
install_tree('bin', prefix.bin)
install('runHALC.py', prefix.bin)
```
should work.

File: var/spack/repos/builtin/packages/halc/package.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        dos2unix = which('dos2unix')
+        dos2unix(join_path(self.prefix.bin, 'runHALC.py'))
+
+    def setup_environment(self, spack_env, run_env):
+        run_env.prepend_path('PATH', self.spec['blasr'].prefix.bin)
+        run_env.prepend_path('PATH', self.spec['lordec'].prefix.bin)
+        run_env.prepend_path('PATH', self.prefix.bin.bin)
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/halc/package.py]: If you do the changes above, you should be able to completely remove `setup_environment`.

File: var/spack/repos/builtin/packages/halc/package.py (hunk 2 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/halc/package.py]: Unless you actually have to specify `all`, in which case yo can override `build_targets` as well.

File: var/spack/repos/builtin/packages/halc/package.py (hunk 3 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/halc/package.py]: Once #8289 is merged, you'll be able to `install_tree('.', prefix)`.

File: var/spack/repos/builtin/packages/halc/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/halc/package.py]: Technically, #8639 needs to be solved first. But yes, these should be `type='run'` if they are only needed at run-time.

**Pull request merged**